spec/services/projects/reupload_creation_wizard_artifact_service_spec.rb

```diff
@@ -0,0 +1,251 @@
+  end
+
+  context "when artifact storage is internal (attachment)" do
+    before do
+      project.update(project_creation_wizard_artifact_export_type: "attachment")
+    end
+
+    it "adds the PDF as an attachment to the existing work package" do
+      initial_attachment_count = work_package.attachments.count
+      result = instance.call
+
+      expect(result).to be_success
+      work_package.reload
+      expect(work_package.attachments.count).to eq(initial_attachment_count + 1)
+
+      attachment = work_package.attachments.last
+      date = Date.current.iso8601
+      expect(attachment.content_type).to eq "application/pdf"
+      expect(attachment.filename).to match(/Important_Project_Project_mandate_#{date}_\d+-\d+.pdf/)
+      expect(attachment.author).to eq(current_user)
+    end
+
+    it "returns a successful service result with the attachment" do
+      result = instance.call
+
+      expect(result).to be_success
+      expect(result.errors).to be_empty
+      expect(result.result).to be_a(Attachment)
+    end
+
+    context "when work package already has an attachment" do
+      before do
+        work_package.attachments.create(
+          author: current_user,
+          file: OpenProject::Files.create_uploaded_file(
+            name: "existing_file.pdf",
+            content_type: "application/pdf",
+            content: "old content",
+            binary: true
+          )
+        )
+      end
+
+      it "adds a new attachment without removing the existing one" do
+        initial_count = work_package.attachments.count
+        result = instance.call
+
+        expect(result).to be_success
+        work_package.reload
+        expect(work_package.attachments.count).to eq(initial_count + 1)
+      end
+    end
+  end
+
+  context "when artifact storage is project storage (file link)" do
+    let(:storage) { create(:nextcloud_storage_with_local_connection) }
+    let(:project_storage) { create(:project_storage, project:, storage:, project_folder_id: "/project_folder") }
+    let(:service_result) { ServiceResult.success(result: nil) }
+
+    before do
+      project.update(
+        project_creation_wizard_artifact_export_type: "file_link",
+        project_creation_wizard_artifact_export_storage: project_storage.id
+      )
+
+      allow(Storages::UploadFileService)
+        .to receive(:call)
+        .and_return(service_result)
+    end
+
+    it "uploads the artifact to the project storage" do
+      result = instance.call
+
+      expect(result).to be_success
+      work_package.reload
+      expect(work_package.attachments.count).to eq(0)
+
+      date = Date.current.iso8601
+      expect(Storages::UploadFileService)
+        .to have_received(:call)
+        .with(container: work_package,
+              project_storage:,
+              file_path: "project_mandate",
+              filename: /Important_Project_Project_mandate_#{date}_\d+-\d+.pdf/,
```

Comment:
Heads-up: The filename pattern has changed with https://community.openproject.org/notifications/details/69426/activity

```suggestion
              filename: /#{project.identifier}_Project_mandate_#{work_package.status.name}_#{date}_\d+-\d+.pdf/,
```